dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1002 @@
+    resource.created_by = response.get('user', '')
+
+    if 'sourceDisk' in request:
+      if not resource.parent:
+        resource.parent = Resource()
+        resource.parent.resource_name = request.get('sourceDisk', '')
+
+    # When creating machine image from an instance
+    elif 'sourceInstance' in request:
+      if not resource.parent:
+        resource.parent = Resource()
+        resource.parent.resource_name = request.get('sourceInstance', '')
+
+    # When creating a disk image from another disk
+    elif 'sourceSnapshot' in request:
+      if not resource.parent:
+        resource.parent = Resource()
+        resource.parent.resource_name = request.get('sourceSnapshot', '')
+
+    # When creating a new instance, one of
+    # initializeParams.sourceImage or initializeParams.sourceSnapshot or
+    # disks.source is required except for local SSD.
+    if request.get('disks'):
+      for disk in request.get('disks', {}):
+        disk_resource = Resource()
+        disk_resource.project_id = resource.project_id
+        disk_resource.created_by = resource.created_by
+        disk_resource.creator_ip_address = resource.creator_ip_address
+        disk_resource.creator_useragent = resource.creator_useragent
+        disk_resource.zone = resource.zone
+        disk_resource.name = disk.get('deviceName')
+        disk_resource.type = 'gce_disk'
+
+        # When creating an instance from a machine image
+        if 'sourceMachineImage' in request:
+          disk_resource.parent = Resource()
+          disk_resource.parent.resource_name = request.get(
+              'sourceMachineImage', '')
+
+        # Check if we already have the disk in the _resources_dict. If true
+        # then add it to the resource disks list and continue to next disk.
+        matched_resource = self._FindResource(disk_resource.name,
+                                              disk_resource.type,
+                                              disk_resource.zone,
+                                              disk_resource.project_id)
+
+        # pylint: disable=line-too-long
+        if matched_resource:
+          matched_resource.created_by = matched_resource.created_by or resource.created_by
+          matched_resource.creator_ip_address = matched_resource.creator_ip_address or resource.creator_ip_address
+          matched_resource.creator_useragent = matched_resource.creator_useragent or resource.creator_useragent
+          matched_resource.parent = matched_resource.parent or disk_resource.parent
+          resource.disks.append(matched_resource)
+          continue
+        # pylint: enable=line-too-long
+
+        initialize_params = disk.get('initializeParams')
+        if initialize_params:
+          # When instance is created using an existing disk image, machine image
+          # or template
+          if 'sourceImage' in initialize_params:
+            if not disk_resource.parent:
```

Comment:
make if not disk_resource.parent: the outer if condition and below it the three cases to not do any further check if no disk_resource.parent

Reply:
Same as the one before removed the check and initialised anyway